contrib/world_builder/include/world_builder/point.h

```diff
@@ -0,0 +1,167 @@
+
+      /**
+       * Constructor. Constructs a Point from an other Point and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const Point<dim> &point, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a 2d Point from two doubles and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const double x, const double y, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a 3d Point from three doubles and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const double x, const double y, const double z, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Destructor
+       */
+      ~Point();
+
+      Point<dim> operator=(const Point<dim> &point);
+
+      /**
+       * dot product
+       */
+      double operator*(const Point<dim> &point) const;
+
+
+      /**
+       * Multiply the vector with a scalar
+       */
+      Point<dim> operator*(const double scalar) const;
+
+      /**
+       * add two points
+       */
+      Point<dim> operator+(const Point<dim> &point) const;
+
+
+      /**
+       * Substract two points
+       */
+      Point<dim> operator-(const Point<dim> &point) const;
+
+
+
+      /**
+       * Multiply the vector with a scalar
+       */
+      Point<dim> &operator*=(const double scalar);
+
+      /**
+       * add two points
+       */
+      Point<dim> &operator+=(const Point<dim> &point);
+
+
+      /**
+       * substract two points
+       */
+      Point<dim> &operator-=(const Point<dim> &point);
+
+      /**
+       * access index (const)
+       */
+      const double &operator[](const unsigned int index) const;
+
+
+      /**
+       * access index
+       */
+      double &operator[](const unsigned int index);
+
+      /**
+       * access index (const)
+       */
+      const double &operator()(const unsigned int index) const;
+
+      /**
+       * access index
+       */
+      double &operator()(const unsigned int index);
+
+
+      /**
+       * return the internal array which stores the point data.
+       */
+      std::array<double,dim> get_array() const;
+
+
+      /**
+       * returns the coordinate system associated with the data.
+       */
+      CoordinateSystem get_coordinate_system() const;
+
+
+      /**
+      * Computes the L2 norm: sqrt(x_i * x_i + y_i * y_i + z_i * z_i) in 3d.
+      */
+      double norm() const;
+
+
+      /**
+      * Computes the square of the norm, which is the sum of the absolute squares
+      * x_i * x_i + y_i * y_i + z_i * z_i in 3d.
+      */
+      double norm_square() const;
+
+
+    private:
+      std::array<double,dim> point;
+      CoordinateSystem coordinate_system;
```

Comment:
you point class also contains a coordinate system. Maybe a name like "location" or something would be more appropriate?

Reply:
I do not fully understand what you mean. Do you want to rename `CoordinateSystem coordinate_system;` to `CoordinateSystem location;`? That name doesn't make much sense to me.